scripts/process_modis.py

```diff
@@ -0,0 +1,399 @@
+                suffix_for_variable = "masked" if masked else "unmasked"
+                dataset = xr.open_dataset(file_path)
+                if masked:
+                    for var in dataset.data_vars:
+                        dataset[var] = dataset[var].where(mask == 1)
+
+                # remove failed downloads (negative date)
+                dataset = dataset.where(dataset["date"] >= 0, drop=True)
+
+                IWP = dataset.Cloud_Water_Path_Ice_Mean
+                IWP = IWP.where(IWP >= 0)
+
+                IWP_counts = dataset.Cloud_Retrieval_Fraction_Ice_Pixel_Counts
+                IWP_counts = IWP_counts.where(IWP_counts >= 0)
+
+                CF_ice = dataset.Cloud_Retrieval_Fraction_Ice * 0.0001
+                CF_ice = CF_ice.where(CF_ice >= 0)
+
+                CF = dataset.Cloud_Fraction_Mean * 0.0001
+                CF = CF.where(CF >= 0)
+                CF_counts = dataset.Cloud_Fraction_Pixel_Counts
+                CF_counts = CF_counts.where(CF_counts >= 0)
+
+                IWP = IWP.where(IWP >= 0).fillna(0)
+                TIWP = IWP * CF_ice
+
+                if masked:
+                    ds["CF_zonal_mean_masked"] += CF.sum(
+                        dim=["lon", "date"], skipna=True
+                    )
+                    ds["TIWP_zonal_mean_masked"] += TIWP.sum(
+                        dim=["lon", "date"], skipna=True
+                    )
+                    CF_zonal_mean_masked_count += (CF >= 0).count(dim=["lon", "date"])
+                    TIWP_zonal_mean_masked_count += (TIWP >= 0).count(
+                        dim=["lon", "date"]
+                    )
+                else:
+                    ds["CF_zonal_mean_unmasked"] += CF.sum(
+                        dim=["lon", "date"], skipna=True
+                    )
+                    ds["TIWP_zonal_mean_unmasked"] += TIWP.sum(
+                        dim=["lon", "date"], skipna=True
+                    )
+                    CF_zonal_mean_unmasked_count += (CF >= 0).count(dim=["lon", "date"])
+                    TIWP_zonal_mean_unmasked_count += (TIWP >= 0).count(
+                        dim=["lon", "date"]
+                    )
+
+                dataset.close()
+
+        lat_reversed = ds["lat"][::-1]  # lats were wrongly saved, need to be reversed
+
+    ds["CF_zonal_mean_unmasked"] /= CF_zonal_mean_unmasked_count
+    ds["TIWP_zonal_mean_unmasked"] /= TIWP_zonal_mean_masked_count
+    ds["CF_zonal_mean_masked"] /= TIWP_zonal_mean_masked_count
+    ds["TIWP_zonal_mean_masked"] /= TIWP_zonal_mean_masked_count
+
+    ds = ds.assign_coords(lat=lat_reversed)
+
+    return ds
+
+
+def interpolate_mask(mask_ds, original_ds):
+    """
+    Interpolate the mask dataset onto modis dataset coordinates
+    """
+    interpolated_mask = mask_ds.astype(int).interp(
+        lat=original_ds.lat, lon=original_ds.lon, method="nearest"
+    )
+    return interpolated_mask
+
+
+def apply_mask_to_dataset(original_ds, mask):
```

Comment:
I included it in the actual functions, since it only required 2 short lines of code. E.g. line 193 in process_modis.py.
I must have forgotten to remove the original function.